File: java/Hot100/Hash/LongestConsecutive.java
package Hot100.Hash;

import java.util.HashSet;
import java.util.Set;

/**
 * Given an unsorted array of integers nums, return the length of the longest consecutive elements sequence.
 *
 * You must write an algorithm that runs in O(n) time.
 */
public class LongestConsecutive {
    public int longestConsecutive(int[] nums) {
        Set<Integer> nums_set=new HashSet<>();
        for (int num:nums)
            nums_set.add(num);

        int length=0;
        for (int num:nums_set){
            if(!nums_set.contains(num-1)){
                int currentNum=num;
                int currentLength=1;

                while(nums_set.contains(++currentNum)){
                    currentLength++;
                }
                length=Math.max(length,currentLength);
            }
        }
        return length;
    }
}

File: python/hash/longest_consecutive.py
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        num_set = set(nums)
        longest = 0

        for num in num_set:
            if num - 1 not in num_set:  # Check if it's the start of a sequence
                curr = num
                length = 1

                while curr + 1 in num_set:
                    curr += 1
                    length += 1

                longest = max(longest, length)

        return longest